### PR TITLE
Test cleanup and refactoring.

### PR DESCRIPTION
**Pull request by bishoprook**
Test cleanup and refactoring.  Created a new TestExtensions class to hold common mocking functionality.  Thoughts on this approach?

### CODE REVIEW

File: Mural.Test/AccountSessionTest.cs [hunk 1]
```diff
@@ -24,42 +24,37 @@ namespace Mural.Test
...
-		protected Mock<ILineConsumer> HookUpMockSink()
-		{
-			Mock<ILineConsumer> sink = new Mock<ILineConsumer>();
-			_session.RaiseUserEvent += sink.Object.HandleUserEvent;
-			sink.Object.RaiseResponseEvent += _session.HandleResponseEvent;
-			return sink;
-		}
 		
+		/// <summary>
+		/// After adding a source, events should be passed through successfully in either
```
Comment (jwodicka) [inline Mural.Test/AccountSessionTest.cs:48]: This comment implies that this method tests the bidirectionality of the hookup.

File: Mural.Test/TestExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+using System;
+using Moq;
+
+namespace Mural.Test
+{
+	public static class TestExtensions
+	{
+		public static Mock<IResponseConsumer> WithMockSource(this ILineConsumer session)
```
Comment (jwodicka) [inline Mural.Test/TestExtensions.cs:8]: I'd never seen this syntax before. This is _fabulous_, and gets at a lot of the clarity we're looking for in making the test cases fluent. Thank you!

Comment (bishoprook) [inline Mural.Test/TestExtensions.cs:8]: Yes, I love a good extension method where appropriate.

I've typically heard "fluent" used to describe something more like this:

public static T WithMockSource<T>(this T sink, out Mock<IResponseConsumer>
source)
    where T : ILineConsumer
{
    source = new Mock<IResponseConsumer>();
    sink.AddSource(source.Object);
    return sink;
}

public static T WithMockSink<T>(this T source, out Mock<ILineConsumer> sink)
    where T : IResponseConsumer
{
    sink = new Mock<ILineConsumer>();
    source.RaiseUserEvent += sink.Object.HandleUserEvent;
    sink.Object.RaiseResponseEvent += source.HandleResponseEvent;
    return source;
}

And then the usage would be something like:

Mock<IResponseConsumer> source;
Mock<ILineConsumer> sink;
_session.WithMockSource(out source).WithMockSink(out sink);

Or if you're instantiating, you get to do something like:

Mock<IResponseConsumer> source;
Mock<ILineConsumer> sink;
AccountSession session = new AccountSession(_account).WithMockSource(out
source).WithMockSink(out sink);

The downside is the use of out parameters, which I am generally loathe to
use.  The upside is that with fluent methods that return the (modified)
target object, you can chain together calls as a single statement.  This is
how all the LINQ methods are implemented, for example.

I lean toward keeping these methods the way they are now.  We don't get
enough readability benefit out of the method chaining to counterbalance the
readability penalty of the use of out parameters.

Comment (jwodicka) [inline Mural.Test/TestExtensions.cs:8]: I suppose I've always considered "Fluent" to mean "Making the expression of
the code as close as possible to the domain of the problem."

And I agree, I'd rather not use the out parameters here; I think the current
construct is far more readable.

**Pull request merged**